src/api/objects/ObjectAPI.js

```diff
@@ -555,27 +555,39 @@ export default class ObjectAPI {
   async getTelemetryPath(identifier, telemetryIdentifier) {
     const objectDetails = await this.get(identifier);
     const telemetryPath = [];
-    if (objectDetails.composition && !['folder'].includes(objectDetails.type)) {
-      let sourceTelemetry = objectDetails.composition[0];
+    if (['folder'].includes(objectDetails.type)) {
+      return telemetryPath;
+    }
+
+    let sourceTelemetry = null;
+    let isParentSameAsChild = true;
+    Object.keys(identifier).forEach((key) => {
+      if (identifier[key] !== telemetryIdentifier[key]) {
+        isParentSameAsChild = false;
+      }
+    });
+    if (isParentSameAsChild) {
+      sourceTelemetry = identifier;
+    } else if (objectDetails.composition) {
+      sourceTelemetry = objectDetails.composition[0];
       if (telemetryIdentifier) {
         sourceTelemetry = objectDetails.composition.find(
           (telemetrySource) =>
             this.makeKeyString(telemetrySource) === this.makeKeyString(telemetryIdentifier)
         );
       }
-      const compositionElement = await this.get(sourceTelemetry);
-      if (!['yamcs.telemetry', 'generator'].includes(compositionElement.type)) {
-        return telemetryPath;
-      }
-      const telemetryKey = compositionElement.identifier.key;
-      const telemetryPathObjects = await this.getOriginalPath(telemetryKey);
-      telemetryPathObjects.forEach((pathObject) => {
-        if (pathObject.type === 'root') {
-          return;
-        }
-        telemetryPath.unshift(pathObject.name);
-      });
     }
+    const compositionElement = await this.get(sourceTelemetry);
+    if (!['yamcs.telemetry', 'generator', 'yamcs.aggregate'].includes(compositionElement.type)) {
+      return telemetryPath;
+    }
+    const telemetryPathObjects = await this.getOriginalPath(compositionElement.identifier);
+    telemetryPathObjects.forEach((pathObject) => {
```

Comment:
Should we just use `.map()` here instead?

Reply:
yeah, I could probably do a `reverse()` and `map()` over that